### PR TITLE
Migration: Update title + description on how to update step

### LINKED ISSUE #93302: Step: How would you like to migrate?
### Details

Integrate the "How would you like to migrate?" step into the flow. It will use the existing step from other flows.

Screenshot 2024-08-06 at 19 15 24

This step will always be displayed after the checkout.

### Checklist

- [ ] The new step is integrated into the flow.

### Related

https://github.com/Automattic/wp-calypso/issues/93258

### PR DESCRIPTION
**Pull request by gabrielcaires**
Closes the #[#93302](https://github.com/Automattic/wp-calypso/issues/93302)

## Proposed Changes

* Enable to customize the How to migrate title and subtitle
* Add the `migration-how-to-upgrade` step extending the `site-migration-how-to-upgrade`

## Testing Instructions
* Access the `/setup/hosted-site-migration-v2` flow
* Follow the steps until it arrives on the `/setup/hosted-site-migration-v2/migration-how-to-upgrade`
* Check if the title matches the figma version
* Check if the subtitle matches with the figma version

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/index.tsx [hunk 1]
```diff
@@ -88,6 +95,12 @@ const SiteMigrationHowToMigrate: Step = function ( { navigation } ) {
 		</>
 	);
 
+	const platformText = shouldDisplayHostIdentificationMessage
```
Comment (renatho) [inline client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-how-to-migrate/index.tsx:98]: Good extraction! 🙌

**Review by renatho — APPROVED** [review]
Looks good and works well!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/16483373

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Hi @gabrielcaires, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `There are two ways to get your site migrated.`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16483373).

**Pull request merged**